packages/mds-policy-service/service/validators.ts

```diff
@@ -14,33 +14,126 @@
-import Joi from 'joi'
+import { SchemaValidator } from '@mds-core/mds-schema-validators'
+import {
+  ACCESSIBILITY_OPTIONS,
+  DAYS_OF_WEEK,
+  MICRO_MOBILITY_VEHICLE_EVENTS,
+  MICRO_MOBILITY_VEHICLE_STATES,
+  MODALITIES,
+  RULE_TYPES,
+  TAXI_VEHICLE_EVENTS,
+  TAXI_VEHICLE_STATES,
+  TNC_VEHICLE_EVENT,
+  TNC_VEHICLE_STATE
+} from '@mds-core/mds-types'
 import { PolicyDomainModel, PolicyMetadataDomainModel } from '../@types'
 
+const stringSchema = (options = {}) => ({ type: 'string', ...options })
+const uuidSchema = { type: 'string', format: 'uuid' }
+// Timestamp Schema that ensures milliseconds
+const TimestampSchema = (options = {}) => ({
+  type: 'integer',
+  minimum: 100_000_000_000,
+  maximum: 99_999_999_999_999,
+  ...options
+})
+
+const arraySchema = (items = {}, options = {}) => ({
+  type: 'array',
+  items,
+  ...options
+})
+
+const enumSchema = <T>(enumType: T[]) => ({ type: 'string', enum: enumType })
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const micromobilityStateMap = MICRO_MOBILITY_VEHICLE_STATES.reduce<any>((acc, state) => {
+  acc[state] = {
+    type: 'array',
+    items: { type: 'string', enum: [...MICRO_MOBILITY_VEHICLE_EVENTS] }
+  }
+  return acc
+}, {})
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const tncStateMap = TNC_VEHICLE_STATE.reduce<any>((acc, state) => {
+  acc[state] = arraySchema(enumSchema([...TNC_VEHICLE_EVENT]))
+  return acc
+}, {})
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const taxiStateMap = TAXI_VEHICLE_STATES.reduce<any>((acc, state) => {
+  acc[state] = arraySchema(enumSchema([...TAXI_VEHICLE_EVENTS]))
+  return acc
+}, {})
+
+const stateModalityIfConditionSchema = (constString: string, props: {}) => ({
+  if: { properties: { modality: { type: 'string', const: constString } } },
+  then: {
+    properties: {
+      states: { type: 'object', properties: props, nullable: true }
+    }
+  }
+})
+
 export const { validate: validatePolicyDomainModel, isValid: isValidPolicyDomainModel } =
-  schemaValidator<PolicyDomainModel>(
-    Joi.object<PolicyDomainModel>()
-      .keys({
-        policy_id: Joi.string().uuid().required(),
-        name: Joi.string().required(),
-        description: Joi.string().required(),
-        provider_ids: Joi.array().items(Joi.string().uuid()).allow(null),
-        start_date: Joi.number().integer().required(),
-        end_date: Joi.number().integer().allow(null),
-        prev_policies: Joi.array().items(Joi.string().uuid()).allow(null),
-        rules: Joi.array().required(),
-        publish_date: Joi.number().integer().allow(null)
-      })
-      .unknown(false)
-  )
+  SchemaValidator<PolicyDomainModel>({
+    $id: 'PolicyDomainModel',
+    type: 'object',
+
+    properties: {
+      policy_id: uuidSchema,
+      name: stringSchema(),
+      description: stringSchema(),
+      provider_ids: arraySchema(uuidSchema, { nullable: true }),
```

Comment:
For nullable things, should also probably put `default: null`, take a look at what I did with the validators in the `mds-ingest-service`.

Reply:
for JSONB, this is moot, null attributes are null when not present.